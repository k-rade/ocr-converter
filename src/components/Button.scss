.btn {
  border-radius: 10rem;
  font-weight: 600;
  padding: 0.625rem 2rem;
  outline: none;
  border: none;
  font-family: var(--font);
  background: var(--color-primary-500);
  font-weight: 500;
  font-size: 1rem;
  cursor: pointer;
  white-space: nowrap;
  color: var(--color-black-one);

  &-full-width {
    width: 100%;
  }

  &:disabled {
    cursor: not-allowed;
  }

  &:hover:not(:disabled) {
    background-color: var(--color-primary-600);
  }
}