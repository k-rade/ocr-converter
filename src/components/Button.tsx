import { FiRefreshCw } from "react-icons/fi";

interface Props {
  text: string | JSX.Element;
  onClick: () => void;
  props?: any;
  fullWidth?: boolean;
  isConverting: boolean;
  conversionProgress: number;
}

const Button = ({
  text,
  onClick,
  fullWidth,
  isConverting,
  conversionProgress,
  ...props
}: Props): JSX.Element => (
  <button
    {...props}
    className={`disabled:cursor-not-allowed cursor-pointer hover:bg-blue-600 relative text-md text-white font-medium bg-blue-500 rounded-full py-2.5 px-4 ${
      fullWidth ? "w-full" : ""
    }`}
    type="button"
    onClick={onClick}
  >
    <div className="absolute flex w-16">
      <FiRefreshCw
        className={`absolute top-1 stroke-blue-300 ${
          isConverting ? "animate-spin " : ""
        }`}
      />
      {isConverting && (
        <div className="mt-0.5 ml-6 text-sm">
          {" "}
          {Math.round(conversionProgress * 100)}%
        </div>
      )}
    </div>
    {isConverting ? "Converting" : text}
  </button>
);

export default Button;
