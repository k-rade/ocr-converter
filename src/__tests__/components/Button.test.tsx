import { fireEvent, render } from "@testing-library/react";
import Button from "../../components/Button";

describe("Button", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(<Button text="text" onClick={jest.fn()} />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onClick", (): void => {
    const mockOnClick = jest.fn();
    const { getByText } = render(<Button text="text" onClick={mockOnClick} />);

    fireEvent.click(getByText("text"));

    expect(mockOnClick).toHaveBeenCalled();
  });
});
