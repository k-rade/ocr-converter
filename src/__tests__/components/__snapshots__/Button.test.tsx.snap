// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<body>
  <div>
    <button
      class="disabled:cursor-not-allowed cursor-pointer hover:bg-blue-600 relative text-md text-white font-medium bg-blue-500 rounded-full py-2.5 px-4 "
      type="button"
    >
      <div
        class="absolute flex w-16"
      >
        <svg
          class="absolute top-1 stroke-blue-300 "
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="23 4 23 10 17 10"
          />
          <polyline
            points="1 20 1 14 7 14"
          />
          <path
            d="M3.51 9a9 9 0 0 1 14.85-3.36L23 10M1 14l4.64 4.36A9 9 0 0 0 20.49 15"
          />
        </svg>
      </div>
      text
    </button>
  </div>
</body>
`;
